src/Numeric/Log.hs

```diff
@@ -41,6 +41,8 @@ import Data.Hashable
 import Data.Hashable.Lifted
 import Data.Int
 import Data.List as List hiding (sum)
+import Data.List.NonEmpty (NonEmpty(..))
+import Data.Semigroup
 #if __GLASGOW_HASKELL__ < 710
 import Data.Monoid
```

Comment:
Looks like the only remaining issue is the conflicting `(<>)` imports on old versions of GHC due to this line. I think you can fix it by removing the `Data.Monoid` import entirely, since `Data.Semigroup` should re-export `Monoid(..)` on old GHCs.